Images from backend

BKSalman opened the pull request:
Instead of using R2 URL directly, add an endpoint to get the images for better control on access

BKSalman committed:
send images bytes from backend

BKSalman committed:
send raw bytes from /api/v1/images endpoint

idk why I was sending it with the json. stoopid

BKSalman committed:
add images endpoint url to frontend + check referer on backend

mohad12211 commented on src/s3/routes.rs:
```diff
@@ -0,0 +1,39 @@
+
+    let bytes = state.storage.get_bytes(&image_path).await.map_err(|e| {
+        tracing::error!("failed to get image bytes: {e}");
+        ImagesError::BadRequest
```
This can be 500 (Internal Server Error) if the request fails or 400 (Bad Request) if the path is invalid, maybe add a TODO to differentiate between the two

BKSalman replied:
just handled it in last commit

mohad12211 commented on src/s3/routes.rs:
```diff
@@ -0,0 +1,39 @@
+        ImagesError::BadRequest
+    })?;
+
+    Ok(bytes.to_vec())
```
Does this do an extra allocation?

BKSalman replied:
removed the `to_vec`

mohad12211 reviewed (changes requested):
General review, will test when i get home

BKSalman committed:
get_image route

- better error handling
- add openapi docs to get_image
- add `s3_referer` to configs
